backend/src/api/model/search/mod.rs

```diff
@@ -278,6 +287,55 @@ pub(crate) async fn all_events(user_query: &str, context: &Context) -> ApiResult
     Ok(EventSearchOutcome::Results(SearchResults { items }))
 }
 
+// See `EventSearchOutcome` for additional information.
+#[derive(juniper::GraphQLUnion)]
+#[graphql(Context = Context)]
+pub(crate) enum SeriesSearchOutcome {
+    SearchUnavailable(SearchUnavailable),
+    Results(SearchResults<search::Series>),
+}
+
+pub(crate) async fn all_series(
+    user_query: &str,
+    writable_only: bool,
+    context: &Context,
+) -> ApiResult<SeriesSearchOutcome> {
+    if !writable_only {
+        context.require_moderator()?;
+    }
+
+    // If the user is not admin, build ACL filter: there has to be one user role
+    // inside the event's ACL.
+    let mut filter = String::new();
+    if !context.auth.is_admin() {
+        let action = if writable_only { "write_roles" } else { "read_roles" };
+        let acl_filter = context.auth.roles()
+            .iter()
+            .map(|role| format!("{} = '{}'", action, hex::encode(role)))
+            .collect::<Vec<_>>()
+            .join(" OR ");
+        write!(
+            filter,
+            "{}({})",
+            if filter.is_empty() { "" } else { " AND " },
+            acl_filter,
+        ).unwrap();
```

Comment:
Just a small thing and it might be intentional, but: The same code is used in the `event_search_query` function (line 350-360), so would it make sense to make this a reusable function?